crates/aptos/src/common/types.rs

```diff
@@ -1318,6 +1334,26 @@ impl TransactionOptions {
             gas_unit_price
         };
 
+        // Get sequence number for account
+        let (account, state) = get_account_with_state(&client, sender_address).await?;
+        let sequence_number = account.sequence_number;
+
+        // Retrieve local time, and ensure it's within an expected skew of the blockchain
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map_err(|err| CliError::UnexpectedError(err.to_string()))?
+            .as_secs();
+        let now_usecs = now * US_IN_SECS;
+        if now_usecs > state.timestamp_usecs + ACCEPTED_CLOCK_SKEW_US {
+            return Err(CliError::UnexpectedError(format!("Failed to send transaction, clock is more than {} seconds ahead of the blockchain {}", ACCEPTED_CLOCK_SKEW_US, state.timestamp_usecs / US_IN_SECS )));
+        } else if now_usecs < state.timestamp_usecs - ACCEPTED_CLOCK_SKEW_US {
+            return Err(CliError::UnexpectedError(format!("Failed to send transaction, clock is more than {} seconds behind of the blockchain {}", ACCEPTED_CLOCK_SKEW_US, state.timestamp_usecs / US_IN_SECS )));
+        }
```

Comment:
the problem even with this logic is that in the horrible circumstance that the blockchain lag is 5 seconds, the CLI is now useless. I would just recommend a reactive rather than proactive approach here. We don't know the real current time.

Reply:
I've changed this to only a warning, if you're behind the blockchain.  So, if the blockchain lag is 5 seconds, and you're 5 seconds behind that, we warn you.  That way at least you know you may need to fix your local clock / increase your timeout